source/common/http/header_map_impl.h

```diff
@@ -109,23 +110,19 @@ class HeaderMapImpl : public HeaderMap, NonCopyable {
     const LowerCaseString* key_;
   };
 
-  struct StaticLookupEntry {
-    typedef StaticLookupResponse (*EntryCb)(HeaderMapImpl&);
-
-    EntryCb cb_{};
-    std::array<std::unique_ptr<StaticLookupEntry>, 256> entries_;
-  };
-
   /**
    * This is the static lookup table that is used to determine whether a header is one of the O(1)
    * headers. This uses a trie for lookup time at most equal to the size of the incoming string.
    */
   struct StaticLookupTable {
```

Comment:
@mattklein123 moved the logic to a common template class but kept this wrapper here because it kind of encapsulates the header creation via it's constructor with `ConstSingleton` so add methods are called for all the headers. I thought moving that to some other place( not sure if there any better place to add the entries to trie) may not be as good as this - but if you have any better ideas , I can try.

Reply:
nit: I think you could have `StaticLookupTable` derive from `TrieLookupTable<EntryCb>` and then you don't need to reimplement the methods. Thoughts?